governance/xc_admin/packages/xc_admin_common/src/programs/types.ts

```diff
@@ -0,0 +1,254 @@
+   */
+  PYTH_LAZER,
+}
+
+/**
+ * Human-readable names for program types
+ */
+export const PROGRAM_TYPE_NAMES: Record<ProgramType, string> = {
+  [ProgramType.PYTH_CORE]: "Pyth Core",
+  [ProgramType.PYTH_LAZER]: "Pyth Lazer",
+};
+
+/**
+ * Type for raw price configs
+ */
+export type PriceRawConfig = {
+  next: PublicKey | null;
+  address: PublicKey;
+  expo: number;
+  minPub: number;
+  maxLatency: number;
+  publishers: PublicKey[];
+};
+
+/**
+ * Type for raw product configs
+ */
+export type ProductRawConfig = {
+  address: PublicKey;
+  priceAccounts: PriceRawConfig[];
+  metadata: Product;
+};
+
+/**
+ * Type for raw mapping configs
+ */
+export type MappingRawConfig = {
+  address: PublicKey;
+  next: PublicKey | null;
+  products: ProductRawConfig[];
+};
+
+/**
+ * Overall raw configuration type
+ */
+export type RawConfig = {
+  mappingAccounts: MappingRawConfig[];
+  permissionAccount?: PermissionData;
+};
+
+export type CoreConfigParams = {
+  accounts: Array<{ pubkey: PublicKey; account: AccountInfo<Buffer> }>;
+  cluster: PythCluster;
+};
+
+/**
+ * Lazer-specific configuration type
+ * TODO: Change to actual Lazer config type
+ */
+export type LazerConfig = {
+  programType: ProgramType.PYTH_LAZER;
+  // Make cluster optional since Lazer might not be tied to a specific cluster
+  cluster?: PythCluster;
+  // More generic data source instead of Solana-specific accounts
+  feeds: LazerFeed[];
+  // Additional metadata that might be relevant for Lazer
+  metadata?: Record<string, unknown>;
+};
+
+/**
+ * Parameters for getting Lazer configuration
+ */
+export type LazerConfigParams = {
+  // Instead of requiring Solana accounts, allow any parameters needed
+  endpoint?: string;
+  network?: string;
+  options?: Record<string, unknown>;
+};
+
+/**
+ * Union type for configuration parameters that can vary by program type
+ */
+export type GetConfigParams =
+  | ({
+      programType: ProgramType.PYTH_CORE;
+    } & CoreConfigParams)
+  | ({ programType: ProgramType.PYTH_LAZER } & LazerConfigParams);
+
+/**
+ * Function to get configuration data
+ */
+export type GetConfigFn = (params: GetConfigParams) => ProgramConfig;
+
+/**
+ * Lazer feed configuration
+ * TODO: Change to actual Lazer feed type
+ */
+export type LazerFeed = {
+  id: string;
+  metadata: Record<string, string | number | boolean>;
+  // Add other feed-specific properties as needed
+};
+
+/**
+ * Type for downloadable price account configuration
+ */
+export type DownloadablePriceAccount = {
+  address: string;
+  publishers: string[];
+  expo: number;
+  minPub: number;
+  maxLatency: number;
+};
+
+/**
+ * Type for downloadable product configuration
+ */
+export type DownloadableProduct = {
+  address: string;
+  metadata: Omit<Product, "price_account">;
+  priceAccounts: DownloadablePriceAccount[];
+};
+
+/**
+ * Type for downloadable configuration
+ */
+export type DownloadableConfig = {
+  [symbol: string]: DownloadableProduct;
```

Comment:
Generally we should prefer `Record<string, DownloadableProduct>` (or at least, we should be consistent about how we defined indexed objects, and most of our code uses `Record`), see [here](https://typescript-eslint.io/rules/consistent-indexed-object-style) for details.